[Profiler] Update third_party/kineto submodule hash

aaronenyeshi opened the pull request:
Summary:
Include improvements such as:
- AMD: roctracer crash fix and roctracer external correlations
- NCCL metadata: process group id to process group name
- Complete nanosecond transition for Kineto
- Remove PrivateUse1Type function causing gpu track to be above cpu tracks
- Use relative time and fix gpu user annotation causing events to overlap

Test Plan: CI and Github CI (full suite)

Reviewed By: sraikund16

Differential Revision: D56475055

Pulled By: aaronenyeshi

aaronenyeshi commented:
cc @@fwenguang

davidberard98 reviewed (approved):
LGTM (pending CI)!

malfet reviewed (approved):
LGTM, but please don't merge -f it especially if it will fail profiler tests

aaronenyeshi committed:
[Profiler] Update third_party/kineto submodule hash

Summary:
Include improvements such as:
- AMD: roctracer crash fix and roctracer external correlations
- NCCL metadata: process group id to process group name
- Complete nanosecond transition for Kineto
- Remove PrivateUse1Type function causing gpu track to be above cpu tracks
- Use relative time and fix gpu user annotation causing events to overlap

Test Plan: CI and Github CI (full suite)

Reviewed By: malfet, davidberard98, sraikund16

Differential Revision: D56475055

Pulled By: aaronenyeshi
